swift/ql/src/queries/Security/CWE-327/BrokenCryptoAlgorithm.qhelp

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+     <overview>
+          <p>
+               Using broken or weak cryptographic algorithms can leave data
+               vulnerable to being decrypted or forged by an attacker.
+          </p>
+
+          <p>
+               Many cryptographic algorithms provided by cryptography
+               libraries are known to be weak, or flawed. Using such an
+               algorithm means that encrypted or hashed data is less
+               secure than it appears to be.
+          </p>
+
+          <p>
+               This query alerts on any use of a weak cryptographic algorithm, that is
+               not a hashing algorithm. Use of broken or weak cryptographic hash
+               functions are handled by the
+               <code>swift/weak-sensitive-data-hashing</code> query.
+          </p>
+
+     </overview>
+     <recommendation>
+
+          <p>
+               Ensure that you use a strong, modern cryptographic
+               algorithm, such as AES-128 or RSA-2048.
```

Comment:
```suggestion
               algorithm, such as AES-256 or RSA-2048.
```
I have nothing concrete, it's probably fine, but I'd be a bit nervous to actually _recommend_ AES-128 today.  When the AES algorithms were new, the US government openly stated that:

> The design and strength of all key lengths of the AES algorithm (i.e., 128, 192 and 256) are sufficient to protect classified information up to the SECRET level. TOP SECRET information will require use of either the 192 or 256 key lengths.

(see [wikipedia](https://en.wikipedia.org/wiki/Advanced_Encryption_Standard))

i.e. AES-128 was considered to be strong enough for some uses, but perhaps not by a wide enough margin to be totally confident against a well equipped attacker.  That was in 2001-ish, much more powerful hardware is available for attackers now.